components/autofill/src/db/store.rs

```diff
@@ -142,14 +148,21 @@ impl StoreImpl {
         })
     }
 
-    /// Creates a store backed by an in-memory database.
+    /// Creates a store backed by an in-memory database with its own memory API (required for unit tests).
     #[cfg(test)]
     pub fn new_memory() -> Self {
         Self {
             db: Mutex::new(crate::db::test::new_mem_db()),
         }
     }
 
+    // Creates a store backed by an in-memory database that shares its memory API (required for autofill sync tests).
+    pub fn new_shared_memory(db_name: &str) -> Result<Self> {
```

Comment:
@mhammond I left the above `test::new_memory` function in place and added this `new_shared_memory` function because the autofill engine unit tests need the separate memory API provided by the `new_mem_db` function while the autofill sync tests need to share the same api. Anyway, let me know if this change makes sense and I'll merge.